### PR TITLE
smarter pickles for Triangulation, VeeringTriangulation and VeeringTriangulationLinearFamily

### LINKED ISSUE #18: Reduce pickle sizes
To be able to compute bigger automata, it would be convenient to reduce the size of pickles.

It is important to know that array have a very bad dumps compared to list of Python integers
```python
>>> import array
>>> from pickle import dumps
>>> dumps([0,1,3,2,-4])
b'\x80\x04\x95\x12\x00\x00\x00\x00\x00\x00\x00]\x94(K\x00K\x01K\x03K\x02J\xfc\xff\xff\xffe.'
>>> dumps(array.array('i', [0,1,3,2,-4]))
b'\x80\x04\x95V\x00\x00\x00\x00\x00\x00\x00\x8c\x05array\x94\x8c\x14_array_reconstructor\x94\x93\x94(\x8c\x05array\x94\x8c\x05array\x94\x93\x94\x8c\x01i\x94K\x08C\x14\x00\x00\x00\x00\x01\x00\x00\x00\x03\x00\x00\x00\x02\x00\x00\x00\xfc\xff\xff\xff\x94t\x94R\x94.'
```

### PR DESCRIPTION
**Pull request by videlec**
Before
```
sage: from veerer import *
sage: print(len(dumps(Triangulation("(0,1,2)"))))
160
sage: print(len(dumps(VeeringTriangulation("(0,1,2)", "RBB"))))
184
sage: print(len(dumps(VeeringTriangulationLinearFamily("(0,1,2)", "RBB", [(1, 1, 0)]))))
649
```
after
```
sage: from veerer import *
sage: print(len(dumps(Triangulation("(0,1,2)"))))
58
sage: print(len(dumps(VeeringTriangulation("(0,1,2)", "RBB"))))
69
sage: print(len(dumps(VeeringTriangulationLinearFamily("(0,1,2)", "RBB", [(1, 1, 0)]))))
127
```

Fixes #18

### CODE REVIEW

File: veerer/linear_family.py [hunk 1]
```diff
@@ -213,6 +220,69 @@ def __init__(self, *args, mutable=False, check=True):
...
+        elif R is QQ:
+            entries = []
+            for i in range(0, len(raw_entries), 2):
+                entries.append(QQ((raw_entries[i], raw_entries[i + 1])))
+        elif _NumberFields is not None and R in _NumberFields:
+            entries = []
+            d = R.degree()
+            for i in range(0, len(raw_entries), d + 1):
+                v = tuple(x / raw_entries[i + d] for x in raw_entries[i:i + d])
+                entries.append(R(v))
```
Comment (saraedum) [inline veerer/linear_family.py:281]: Isn't there an `else: entries = raw_entries` missing?

**Review by saraedum — APPROVED** [review]

File: veerer/linear_family.py (hunk 1 above)
Comment (videlec) [reply veerer/linear_family.py:281]: Don't think so, there is a `entries = []` before the loop and then it is populated from `raw_entries`.

Comment (videlec) [reply veerer/linear_family.py:281]: I will at least add a doctest to it!

Comment (videlec) [reply veerer/linear_family.py:291]: I understood what you meant :-) (but I do not know how to test it though)

Comment (videlec) [reply veerer/linear_family.py:291]: fixed in 2c3184a. Thanks.

**Pull request merged**